src/datasets/builder.py

```diff
@@ -570,18 +578,23 @@ def get_imported_module_dir(cls):
 
     def download_and_prepare(
         self,
+        output_dir: Optional[str] = None,
         download_config: Optional[DownloadConfig] = None,
         download_mode: Optional[DownloadMode] = None,
         ignore_verifications: bool = False,
         try_from_hf_gcs: bool = True,
         dl_manager: Optional[DownloadManager] = None,
         base_path: Optional[str] = None,
         use_auth_token: Optional[Union[bool, str]] = None,
+        file_format: Optional[str] = None,
+        storage_options: Optional[dict] = None,
         **download_and_prepare_kwargs,
     ):
         """Downloads and prepares dataset for reading.
 
         Args:
+            output_dir (:obj:`str`, optional): output directory for the dataset.
+                Default to this builder's ``cache_dir``, which is inside ~/.cache/huggingface/datasets by default.
```

Comment:
We might add the docs directive:
```
<Added version="2.5.0"/>
```

The same for the other new params in this docstring (`file_format` and `storage_options`, besides `output_dir`) and the `fs` params in the `DatasetInfo` docstring.